sw/host/tests/chip/rv_dm/src/mem_access.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::PathBuf;
+
+use anyhow::{ensure, Result};
+use clap::Parser;
+use rand::prelude::*;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::execute_test;
+use opentitanlib::io::jtag::JtagTap;
+use opentitanlib::test_utils::init::InitializeTest;
+
+use bindgen::dif;
+use top_earlgrey::top_earlgrey;
+
+#[derive(Debug, Parser)]
+struct Opts {
+    #[command(flatten)]
+    init: InitializeTest,
+
+    /// Path to the ROM binary.
+    #[arg(long)]
+    rom: PathBuf,
+}
+
+const NUM_ACCESSES_PER_REGION: usize = 32;
+
+fn test_mem_access(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    // Avoid watchdog timeout by entering bootstrap mode.
+    transport.pin_strapping("ROM_BOOTSTRAP")?.apply()?;
+
+    transport.pin_strapping("PINMUX_TAP_RISCV")?.apply()?;
+    transport.reset_target(opts.init.bootstrap.options.reset_delay, true)?;
+
+    let jtag = &mut *opts
+        .init
+        .jtag_params
+        .create(transport)?
+        .connect(JtagTap::RiscvTap)?;
+
+    let rom_data = std::fs::read(&opts.rom)?;
+    ensure!(rom_data.len() % 4 == 0);
+
+    let rw_regions = [
+        (
+            "ram_ret",
+            top_earlgrey::RAM_RET_AON_BASE_ADDR as u32,
+            top_earlgrey::RAM_RET_AON_SIZE_BYTES as u32,
+        ),
+        (
+            "ram_main",
+            top_earlgrey::RAM_MAIN_BASE_ADDR as u32,
+            top_earlgrey::RAM_MAIN_SIZE_BYTES as u32,
+        ),
+        (
+            "otbn_imem",
+            top_earlgrey::OTBN_BASE_ADDR as u32 + dif::OTBN_IMEM_REG_OFFSET,
+            dif::OTBN_IMEM_SIZE_BYTES,
+        ),
+        (
+            "otbn_dmem",
+            top_earlgrey::OTBN_BASE_ADDR as u32 + dif::OTBN_DMEM_REG_OFFSET,
+            dif::OTBN_DMEM_SIZE_BYTES,
+        ),
+    ];
+
+    // Randomly generate accesses.
+    let mut rng = thread_rng();
```

Comment:
Can you use a seed and print it every run so that we can reproduce CI failures locally? 
We can also add an optional argument for the seed and then generated a randon seed only if not provided.

Reply:
Printing the seed sounds like a good idea, otherwise it's hard to reproduce failures